synapse/config/cas.py

```diff
@@ -41,6 +41,7 @@ def read_config(self, config: JsonDict, **kwargs: Any) -> None:
             public_baseurl = self.root.server.public_baseurl
             self.cas_service_url = public_baseurl + "_matrix/client/r0/login/cas/ticket"
 
+            self.cas_protocol_version = cas_config.get("protocol_version")
```

Comment:
We could give more obvious errors when an unsupported protocol version is specified:

```suggestion
            if self.cas_protocol_version not in [1, 2, 3]:
                raise ConfigError(
                    "Unsupported CAS protocol version %s (only versions 1, 2, 3 are supported)"
                    % (self.cas_protocol_version,),
                    ("cas_config", "protocol_version"),
                )
```